src/chainstate/stacks/boot/costs-2.clar

```diff
@@ -1 +1,567 @@
+    (runtime (logn n u1 u2)))
+
+(define-read-only (cost_analysis_check_tuple_merge (n uint))
+    (runtime (linear n u1000 u1000)))
+
+(define-read-only (cost_analysis_check_tuple_cons (n uint))
+    (runtime (nlogn n u3 u5)))
+
+(define-read-only (cost_analysis_tuple_items_check (n uint))
+    (runtime (linear n u1 u59)))
+
+(define-read-only (cost_analysis_check_let (n uint))
+    (runtime (linear n u1 u12)))
+
+(define-read-only (cost_analysis_lookup_function (n uint))
+    (runtime u20))
+
+(define-read-only (cost_analysis_lookup_function_types (n uint))
+    (runtime (linear n u1 u28)))
+
+(define-read-only (cost_analysis_lookup_variable_const (n uint))
+    (runtime u15))
+
+(define-read-only (cost_analysis_lookup_variable_depth (n uint))
+    (runtime (nlogn n u1 u34)))
+
+(define-read-only (cost_ast_parse (n uint))
+    (runtime (linear n u172 u287441)))
+
+(define-read-only (cost_ast_cycle_detection (n uint))
+    (runtime (linear n u141 u72)))
+
+(define-read-only (cost_analysis_storage (n uint))
+    {
+        runtime: (linear n u2 u100),
+        write_length: (linear n u1 u1),
+        write_count: u1,
+        read_count: u1,
+        read_length: u1
+    })
+
+(define-read-only (cost_analysis_use_trait_entry (n uint))
+    {
+        runtime: (linear n u9 u723),
+        write_length: (linear n u1 u1),
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_analysis_get_function_entry (n uint))
+    {
+        runtime: (linear n u81 u1303),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+
+(define-read-only (cost_analysis_fetch_contract_entry (n uint))
+    {
+        runtime: (linear n u1000 u1000),
+        write_length: u0,
+        write_count: u0,
+        read_count: u1,
+        read_length: (linear n u1 u1)
+    })
+
+(define-read-only (cost_lookup_variable_depth (n uint))
+    (runtime (linear n u2 u14)))
+
+(define-read-only (cost_lookup_variable_size (n uint))
+    (runtime (linear n u2 u1)))
+
+(define-read-only (cost_lookup_function (n uint))
+    (runtime u16))
+
+(define-read-only (cost_bind_name (n uint))
+    (runtime u256))
+
+(define-read-only (cost_inner_type_check_cost (n uint))
+    (runtime (linear n u2 u9)))
+
+(define-read-only (cost_user_function_application (n uint))
+    (runtime (linear n u26 u140)))
+
+(define-read-only (cost_let (n uint))
+    (runtime (linear n u146 u862)))
+
+(define-read-only (cost_if (n uint))
+    (runtime u200))
+
+(define-read-only (cost_asserts (n uint))
+    (runtime u158))
+
+(define-read-only (cost_map (n uint))
+    (runtime (linear n u1210 u3314)))
+
+(define-read-only (cost_filter (n uint))
+    (runtime u460))
+
+(define-read-only (cost_len (n uint))
+    (runtime u486))
+
+(define-read-only (cost_element_at (n uint))
+    (runtime u619))
+
+(define-read-only (cost_index_of (n uint))
+    (runtime (linear n u1 u243)))
```

Comment:
I know that most Clarity code that uses the VRF or does some kind of parsing makes heavy use of `index-of` to convert 1-byte buffers into uints.  Since proper slicing and conversion methods won't be available until 2.1, is there a world in which we just declare that this function has O(1) cost, since lists are finite length anyway and we could always implement lists internally as hashmaps or some such if it became a problem?

Reply:
I'm not sure what the benefit would be of doing so -- is it to just make this operation cheaper than we're measuring?